### ISSUE TITLE
Traefik v1.7 misses events from k8s, does not serve services

### ISSUE THREAD

**Issue by rkojedzinszky** [containous/traefik#6626]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Just running traefik as an ingress controller on Kubernetes.

### What did you expect to see?

Expect to route traffic to all ready backends/pods.

### What did you see instead?

Some ready pods were not treated by Traefik, and thus, was not routed traffic to. Meanwhile observed `Skipping same configuration for provider` messages.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.7.24
Codename:     maroilles
Go version:   go1.14.1
Built:        2020-03-25_04:34:11PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Kubernetes deployment.

### Hints

I found that, the mentioned message comes from https://github.com/containous/traefik/blob/v1.7/server/server_configuration.go#L324. If a pod goes to a NotReady state, it generates an event for traefik. It starts processing it, but if recovery is too fast (i.e. the new event comes in while still processing the first one) the code pointed finds that existing and new configuration are equal, thus skipping. Later, processing the first event finishes, resulting in the service brought and remaining in down state from Traefik's perspective.

Perhaps even during the --providersthrottleduration interval such a recovery event will be dropped.

On my environment, reloading the configuraion (the function at https://github.com/containous/traefik/blob/v1.7/server/server_configuration.go#L34) sometimes takes a few seconds, so it is very easy to reproduce the issue. I have a pod with 1 sec readinessProbe, and just need to flap its' readiness down and up, thus Traefik receives two events, with the latter being dropped.

**Comment by rkojedzinszky** [comment]
In my lab environment, I've compiled Traefik without the `same configuration` check, and seems that with that I cannot reproduce the issue. I dont know what other side-effects may it have, but at least Traefik gets into the desired state sooner or later.

**Comment by traefiker** [comment]
Closed by #6631.
